File: 1460-make-two-arrays-equal-by-reversing-subarrays/1460-make-two-arrays-equal-by-reversing-subarrays.go
func canBeEqual(target []int, arr []int) bool {
    freq := make([]int, 1001)
    for _, num := range arr {
        freq[num]++
    }

    for _, num := range target {
        freq[num]--
        if freq[num] == -1 {
            return false
        }
    }
    
    return true
}

File: 1460-make-two-arrays-equal-by-reversing-subarrays/1460-make-two-arrays-equal-by-reversing-subarrays.java
class Solution {
    public boolean canBeEqual(int[] target, int[] arr) {
        var freq = new int[1001];
        for(var num : arr) freq[num]++;

        for(var num : target){
            freq[num]--;
            if(freq[num] == -1) return false;
        }
        return true;
    }
}